promql/engine.go

```diff
@@ -15,6 +15,7 @@ package promql
 
 import (
 	"container/heap"
+	native_context "context"
```

Comment:
Can we get rid of this by finally migrating?

Reply:
Here I can just change it without problems actually, but in `web/api/v2/api.go` the generated protobufs still expect old contexts. I managed to install protoc at the required version and all that, but it still generates the non-native contexts. I found https://github.com/golang/protobuf/issues/217 which sounds like that isn't going to be changed very soon... so I just changed this occurrence here for now and converted the engine tests as well. Seems ok? I'm happy to convert other non-native contexts too, but it seems it would be better to put into a separate PR.

Reply:
Yea, fine with me, it's just an itch. Maybe the go1.9 type aliases can solve it once we make it our minimum version